reference-implementation/lib/writable-stream.js

```diff
@@ -188,7 +188,14 @@ function WritableStreamError(stream, e) {
 }
 
 function WritableStreamFinishClose(stream) {
-  assert(stream._state === 'closing' || stream._state === 'errored');
+  assert(stream._pendingCloseRequest !== undefined);
+  stream._pendingCloseRequest._resolve(undefined);
+  stream._pendingCloseRequest = undefined;
+
+  if (stream._pendingAbortRequest !== undefined) {
```

Comment:
Sorry, the diff confused me.

It changes the order that the abort() promise is resolved wrt. the .closed promise. This is observable in principle. When I tried to write tests for promise resolve ordering, the results were not what I expected, so I never added any. Probably no-one cares.